### DISCUSSION TITLE
Electrum wallet to Raspiblitz connection on 50002 or 50001?

### DISCUSSION THREAD

**Discussion by Nerfzooka** [romanz/electrs#1012] [Q&A]
Hi electrs fans, let me know if this question is out of place, but my Raspiblitz always used to connect on the local network with port 50002.  It got a corrupted memory so I had to make a fresh install and now all is assembled, synced, indexed and working as before, but the old 50002 port connection doesn't work anymore.  I changed to 50001:t and it does connect.  I wonder what's happening there.

**Comment by romanz** [comment]
CC: @rootzoll @openoms

**Comment by romanz** [comment]
@Nerfzooka could you please attach electrs' logs?

**Comment by Nerfzooka** [comment]
Sure. Here are the logs I know about.  Let me know if there are others.
_____________________________
STATUS ELECTRS SERVICE

version='v0.10.0'
configured=1
serviceInstalled=1
serviceRunning=1
localIP='REDACTED'
publicIP='REDACTED'
portTCP='50001'
localTCPPortActive=1
publicTCPPortAnswering=0
portSSL='50002'
localHTTPPortActive=1
publicHTTPPortAnswering=0
TorRunning=1
nginxTest=1
serviceRunning=1
electrumResponding=1
initialSynced=1

_____________________________
**"sudo journalctl -u electrs" output**

-- Journal begins at Sun 2024-02-25 04:12:28 GMT, ends at Thu 2024-02-29 19:18:31 GMT. --
Feb 26 15:35:42 raspberrypi systemd[1]: Started Electrs.
Feb 26 15:35:42 raspberrypi electrs[3829169]: Starting electrs 0.10.0 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/app-storage/electrs/db/bitcoin", daemon_dir: "/home/electrs/.bitcoin", daemon_auth: UserPass("raspibolt", ""), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs v0.10.0 - the Electrum Rust Server on your RaspiBlitz", signet_magic: f9beb4d9, args: [] }
Feb 26 15:51:14 raspberrypi electrs[3829169]: [2024-02-26T15:51:14.421Z ERROR electrs::server] 9: disconnecting due to failed to send response: "{"error":{"code":-32603,"message":"unavailable index"},"id":"electrs-check","jsonrpc":"2.0"}\n"
Feb 26 15:51:14 raspberrypi electrs[3829169]: [2024-02-26T15:51:14.422Z WARN electrs::server] 9: failed to shutdown TCP connection Transport endpoint is not connected (os error 107)
Feb 26 17:35:22 raspberrypi electrs[3829169]: [2024-02-26T17:35:22.690Z ERROR electrs::server] 493: disconnecting due to failed to send response: "{"error":{"code":-32603,"message":"unavailable index"},"id":"electrs-check","jsonrpc":"2.0"}\n"
Feb 26 17:35:22 raspberrypi electrs[3829169]: [2024-02-26T17:35:22.691Z WARN electrs::server] 493: failed to shutdown TCP connection Transport endpoint is not connected (os error 107)
Feb 27 17:00:56 raspberrypi electrs[3829169]: Error: electrs failed
Feb 27 17:00:56 raspberrypi electrs[3829169]: Caused by:
Feb 27 17:00:56 raspberrypi electrs[3829169]: 0: sync failed
Feb 27 17:00:56 raspberrypi electrs[3829169]: 1: sending on a disconnected channel
Feb 27 17:00:58 raspberrypi systemd[1]: electrs.service: Main process exited, code=exited, status=1/FAILURE
Feb 27 17:00:58 raspberrypi systemd[1]: electrs.service: Failed with result 'exit-code'.
Feb 27 17:00:58 raspberrypi systemd[1]: electrs.service: Consumed 21h 11min 8.247s CPU time.
Feb 27 17:01:58 raspberrypi systemd[1]: electrs.service: Scheduled restart job, restart counter is at 1.
Feb 27 17:01:58 raspberrypi systemd[1]: Stopped Electrs.
Feb 27 17:01:58 raspberrypi systemd[1]: electrs.service: Consumed 21h 11min 8.247s CPU time.
Feb 27 17:02:39 raspberrypi systemd[1]: Started Electrs.
Feb 27 17:02:40 raspberrypi electrs[2594922]: Starting electrs 0.10.0 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/app-storage/electrs/db/bitcoin", daemon_dir: "/home/electrs/.bitcoin", daemon_auth: UserPass("raspibolt", ""), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs v0.10.0 - the Electrum Rust Server on your RaspiBlitz", signet_magic: f9beb4d9, args: [] }
Feb 27 17:04:26 raspberrypi electrs[2594922]: Error: electrs failed
Feb 27 17:04:26 raspberrypi electrs[2594922]: Caused by:
Feb 27 17:04:26 raspberrypi electrs[2594922]: 0: sync failed
Feb 27 17:04:26 raspberrypi electrs[2594922]: 1: sending on a disconnected channel
Feb 27 17:04:26 raspberrypi systemd[1]: electrs.service: Main process exited, code=exited, status=1/FAILURE
Feb 27 17:04:26 raspberrypi systemd[1]: electrs.service: Failed with result 'exit-code'.
Feb 27 17:04:26 raspberrypi systemd[1]: electrs.service: Consumed 14.589s CPU time.
Feb 27 17:04:32 raspberrypi systemd[1]: Stopped Electrs.
Feb 27 17:04:32 raspberrypi systemd[1]: electrs.service: Consumed 14.589s CPU time.
-- Boot 6afc180879904ba0afcb8730084f50f4 --
Feb 27 17:07:38 raspberrypi systemd[1]: Started Electrs.
Feb 27 17:07:39 raspberrypi electrs[12277]: Starting electrs 0.10.0 on aarch64 linux with Config { network: Bitcoin, db_path: "/mnt/hdd/app-storage/electrs/db/bitcoin", daemon_dir: "/home/electrs/.bitcoin", daemon_auth: UserPass("raspibolt", ""), daemon_rpc_addr: 127.0.0.1:8332, daemon_p2p_addr: 127.0.0.1:8333, electrum_rpc_addr: 0.0.0.0:50001, monitoring_addr: 127.0.0.1:4224, wait_duration: 10s, jsonrpc_timeout: 15s, index_batch_size: 10, index_lookup_limit: None, reindex_last_blocks: 0, auto_reindex: true, ignore_mempool: false, sync_once: false, skip_block_download_wait: false, disable_electrum_rpc: false, server_banner: "Welcome to electrs v0.10.0 - the Electrum Rust Server on your RaspiBlitz", signet_magic: f9beb4d9, args: [] }
Feb 28 00:56:41 raspberrypi electrs[12277]: [2024-02-28T00:56:41.274Z WARN electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 03:15:45 raspberrypi electrs[12277]: [2024-02-28T03:15:45.331Z WARN electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 03:22:39 raspberrypi electrs[12277]: [2024-02-28T03:22:39.286Z WARN electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 08:07:49 raspberrypi electrs[12277]: [2024-02-28T08:07:49.043Z WARN electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 09:54:42 raspberrypi electrs[12277]: [2024-02-28T09:54:42.614Z WARN electrs::mempool] mempool sync failed: failed to get mempool txids
Feb 28 22:18:26 raspberrypi electrs[12277]: [2024-02-28T22:18:26.995Z WARN electrs::mempool] mempool sync failed: failed to get mempool txids

**Comment by rootzoll** [comment]
Port 50002 is managed thru the nginx for SSL .. maye something got broken there?
https://github.com/raspiblitz/raspiblitz/blob/003230d1377333341d57568628a5d8024793cfa7/home.admin/config.scripts/bonus.electrs.sh#L412-L413

But so far tested with v1.11.0rc3 and 50002 is working on my testmachine.

**Comment by Nerfzooka** [comment]
Ah, nginx.  The nginx.conf file looks like this:
```
 stream {
        upstream electrs {
                server 127.0.0.1:50001;
        }
        server {
                listen 50002 ssl;
                proxy_pass electrs;
                ssl_certificate /mnt/hdd/app-data/nginx/tls.cert;
                ssl_certificate_key /mnt/hdd/app-data/nginx/tls.key;
                ssl_session_cache shared:SSL-electrs:1m;
                ssl_session_timeout 4h;
                ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
                ssl_prefer_server_ciphers on;


        }
```
The status looks like this:
![nginxStatus - Copy](https://github.com/romanz/electrs/assets/23216537/20f518a3-0306-4784-a7a0-3277a00bf0fb)

Hmmm, Is that last  "journal rotating" line suspicious?

**Comment by Nerfzooka** [comment]
Oh look, a sparrow wallet gives some connection details:

```
Could not connect:

PKIX path validation failed: java.security.cert.CertPathValidatorException: signature check failed

PKIX path validation failed: java.security.cert.CertPathValidatorException: signature check failed
```

I wonder how I can fix a signature.

**Comment by Nerfzooka** [comment]
Hey! Not electrs' fault at all.  Not nginx' fault, not raspiblitz' fault.  
The wallets I've been using for a long time had cert files that matched my old raspiblitz.  When the raspiblitz sd card got too old and began to change read/write files to readonly I had to do a whole fresh install that came with new tls certs.  The new certs didn't match the old certs that the wallets were holding.  Port 50001 connected, 50002 didn't.  The cert directory for each wallet was in ThisPC > Local Disk (C:) > Users > Username > AppData > Roaming > WalletName > certs.  Just change the name of the entry for your local area network and the next time you open your wallet it will create a new cert.

**Reply by JSM684** [reply]
Hello there, I have encountered the same error message but I am afraid I am not sophisticated enough to understand your reply. I have checked the electrum rust server on raspiblitz and it matches the URL in the private electrum server on sparrow (including port), but I am getting the same error:

Could not connect:

PKIX path validation failed: java.security.cert.CertPathValidatorException: signature check failed

PKIX path validation failed: java.security.cert.CertPathValidatorException: signature check failed

I have gone to the cert directory for the wallert and it also matches. I am not sure how to proceed with the remedy you suggested  "change the name of the entry for your local area network and the next time you open your wallet it will create a new cert."

**Reply by Nerfzooka** [reply]
The wallet may be using an old certificate. Instead of looking for a match, delete the match (or move it or rename it), and the wallet will make a new cert.

**Reply by JSM684** [reply]
Many thanks - confirming I deleted the certificate and was able to reconnect.
